src/CRM/CivixBundle/Command/AddDaosCommand.php

```diff
@@ -0,0 +1,104 @@
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use CRM\CivixBundle\Builder\Collection;
+use CRM\CivixBundle\Builder\Dirs;
+use CRM\CivixBundle\Builder\Info;
+use CRM\CivixBundle\Builder\PhpData;
+use CRM\CivixBundle\Builder\Template;
+use CRM\CivixBundle\Utils\Path;
+use Exception;
+
+class AddDaosCommand extends \Symfony\Component\Console\Command\Command {
+  const API_VERSION = 3;
+
+  protected function configure() {
+    $this
+      ->setName('generate:daos')
+      ->setDescription('Create DAOs and mysql schema based on this extensions xml schema definition files (*EXPERIMENTAL AND INCOMPLETE*)')
+      ->setHelp(
+        "Create DAOs and mysql schema based on this extensions xml schema definition files\n" .
+        "\n" .
+        "Typically, one will have created / updated on or more xml/CRM/NameSpace/EntityName.xml files\n" .
+        "before running this command.\n"
+      );
+  }
+
+  protected function execute(InputInterface $input, OutputInterface $output) {
+    // load Civi to get access to civicrm_api_get_function_name
+    Services::boot(['output' => $output]);
+    $civicrm_api3 = Services::api3();
+    if (!$civicrm_api3 || !$civicrm_api3->local) {
+      $output->writeln("<error>Require access to local CiviCRM source tree. Configure civicrm_api3_conf_path.</error>");
+      return;
+    }
+
+    $ctx = [];
+    $ctx['type'] = 'module';
+    $ctx['basedir'] = \CRM\CivixBundle\Application::findExtDir();
+    $basedir = new Path($ctx['basedir']);
+
+    $info = new Info($basedir->string('info.xml'));
+    $info->load($ctx);
+    $attrs = $info->get()->attributes();
+    if ($attrs['type'] != 'module') {
+      $output->writeln('<error>Wrong extension type: ' . $attrs['type'] . '</error>');
+      return;
+    }
+
+    $xmlSchemaGlob = "xml/schema/{$ctx['namespace']}/*.xml";
+    $absXmlSchemaGlob = $basedir->string($xmlSchemaGlob);
+    // var_dump(($absXmlSchemaGlob));
+    $xmlSchemas = glob($absXmlSchemaGlob);
+
+    if(!count($xmlSchemas)){
+      throw new Exception("Could not find files matching '$xmlSchemaGlob'. You may want to run `civix generate:entity` before running this command.");
+    }
+
+    $specification = new \CRM_Core_CodeGen_Specification;
+    $specification->buildVersion = \CRM_Utils_System::majorVersion();
+
+    $config = new \stdClass;
+    $config->phpCodePath = $basedir->string('');
+    $config->sqlCodePath = $basedir->string('sql/');
+
+    foreach($xmlSchemas as $xmlSchema){
+
+      $dom = new \DomDocument();
+      $xmlString = file_get_contents($xmlSchema);
+      $dom->loadXML($xmlString);
+      $xml = simplexml_import_dom($dom);
+
+      $specification->getTable($xml, $database, $tables);
+
+      $tables[(string) $xml->name]['sourceFile'] = $xmlSchema;
+      $config->tables = $tables;
+
+      $dao = new \CRM_Core_CodeGen_DAO($config, (string) $xml->name);
```

Comment:
I haven't `r-run` this PR yet, but -- based on skimming -- this looks like the real deal for generating DAOs in `civix`. Much nicer than hybrid bash/manual workflows we've used before...